src/valuetypes.ts

```diff
@@ -530,20 +542,30 @@ export async function guessValueType(editors: ValueTypeEditor[], name: string, i
   return results[0].editor;
 }
 
-export function createTypeEditor(editors: ValueTypeEditor[], current: ValueTypeEditor, emptyOne = true, idType: string = '') {
-  return `
-  <select class='form-control'>
-          ${emptyOne? '<option value=""></option>':''}
-          ${editors.map((editor) => `<option value="${editor.id}" ${current && current.id === editor.id ? 'selected="selected"' : ''}>${editor.name} ${current && current.id === editor.id && current.name === 'IDType' ? `(${idType})` : ''}</option>`).join('\n')}
-        </select>
-        <span class="input-group-btn">
-          <button class="btn btn-secondary" ${!current || !current.hasEditor ? 'disabled="disabled' : ''} type="button"><i class="glyphicon glyphicon-cog"></i></button>
-        </span>`;
+export function createTypeEditor(editors: ValueTypeEditor[], current: ValueTypeEditor, def: ITypeDefinition, emptyOne = true) {
+  const options = editors.map((editor) => editor.getOptionsMarkup(current, def)).join('\n');
+
+ return `<select class="form-control">
+        ${emptyOne? '<option value=""></option>':''}
+        ${options}
+    </select>
+    <span class="input-group-btn">
+      <button class="btn btn-secondary" ${!current || !current.hasEditor ? 'disabled="disabled' : ''} type="button"><i class="glyphicon glyphicon-cog"></i></button>
+    </span>`;
 }
 
 export function updateType(editors: ValueTypeEditor[], emptyOne = true) {
   return function (d) {
-    const type = ((emptyOne && this.selectedIndex <= 0)) ? null : editors[this.selectedIndex < 0 ? 0 : this.selectedIndex - (emptyOne?1:0)];
+    const parent = this.options[this.selectedIndex].parentNode;
+
+    let type = null;
+    if(parent.nodeName !== 'OPTGROUP') {
+      type = editors.find((editor) => editor.id === this.value);
+    } else{
+      type = editors.find((editor) => editor.id === parent.dataset.type);
+      d.value[parent.dataset.type] = this.value;
```

Comment:
here I check if an element is inside an <optgroup>:
 - if not: set type to the editor, which was selected with this.value
 - if yes: set the type to the editor, which has the data-type attribute (e.g. idType)
      -- set the idType to the value, which editor was chosen (Ensembl, Cellline, Tissue)

is there another way to set the idType (subtype) of the corresponding IDTypeEditor?